File: solutions/1. Two Sum/1.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    h = {}
    nums.each_with_index do |value, i|
        n = target - value
        if h[n]  
            return [h[n], i]
        else
            h[value] = i
        end
    end
end

File: solutions/1. Two Sum/1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        h = {}
        for i in range(len(nums)):
            n = target - nums[i]
            if n in h:
                return [h[n], i]
            else:
                h[nums[i]] = i